prow/cmd/branchprotector/README.md

```diff
@@ -0,0 +1,168 @@
+
+branchprotector configures [github branch protection] according to a specified
+policy.
+
+## Policy configuration
+
+Extend the primary prow [`config.yaml`] document to include a top-level
+`branch-protection` key that looks like the following:
+
+```yaml
+
+branch-protection:
+  orgs:
+    kubernetes:
+      repos:
+        test-infra:
+          # Protect all branches in kubernetes/test-infra
+          protect-by-default: true
+          # Always allow the org's oncall-team to push
+          pushers: ["oncall-team"]
+          # Ensure that the extra-process-followed github context passes
+          # In addition, adds any always_run: True prow jobs
+          require-contexts: ["extra-process-followed"]
+
+presubmits:
+  kubernetes/test-infra:
+  - name: fancy-job-name
+    context: fancy-job-name
+    always_run: true
+    spec:  # podspec that runs job
+```
+
+This config will:
+  * Enable protection for every branch in the `kubernetes/test-infra`
+repo.
+  * Require `extra-process-followed` and `fancy-job-name` contexts to pass
+    before allowing a merge
+    - Although it will always allow `oncall-team` to merge, even if required
+      contexts fail.
+    - Note that `fancy-job-name` is pulled in automatically from the
+      `presubmits` config for the repo, if one exists.
+
+### Updating
+
+* Send PR with `config.yaml` changes
+* Merge PR
+* Done!
+
+Make changes to the policy by modifying [`config.yaml`] in your favorite text
+editor and then send out a PR. When the PR merges prow pushes the updated config
+. The next time the branch protection job runs it will apply the new policies.
+
+### Advanced configuration
+
+See [`branch_protection.go`] for a complete list of fields allowed
+inside `branch-protection`.  It is possible to define a policy at the
+`branch-protection`, `org`, `repo` or `branch` level. For example:
+
+```yaml
+branch-protection:
+  # Protect unless overridden
+  protect-by-default: true
+  # If protected, always require the cla context
+  require-contexts: ["cla"]
+  orgs:
+    # This
+    unprotected-org:
+      # Disable protection unless overridden (overrides parent setting of true)
+      protect-by-default: false
+      repos:
+        protected-repo:
+          # Inherit protect-by-default config from parent
+          # If protected, always require the tested context
+          require-contexts: ["tested"]
+          branches:
+            secure:
+              # Protect the secure branch (overrides inhereted parent setting of false)
+              protect-by-default: true
+              require-contexts: ["foo"]
+    different-org:
+      protect-by-default: null
```

Comment:
would you expect someone to do this or just omit it?

Reply:
Done